nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GenerateTableFetch.java

```diff
@@ -115,20 +128,36 @@ public GenerateTableFetch() {
 
     @OnScheduled
     public void setup(final ProcessContext context) {
+        // The processor is invalid if there is an incoming connection and max-value columns are defined
+        if (context.getProperty(MAX_VALUE_COLUMN_NAMES).isSet() && context.hasIncomingConnection()) {
+            throw new ProcessException("If an incoming connection is supplied, no max-value column names may be specified");
```

Comment:
The other concern was if we support Max Value columns using attributes, we'd have to keep state for those columns for each table, which could get unwieldy for state management depending on the number of columns and tables. Also if I add this logic, I'll need to change how state is stored (to change the key to a full-qualified column name), which could impact backwards compatibility (if state is maintained across an upgrade).